docs/writing-benchmarks.md

```diff
@@ -0,0 +1,236 @@
+# Writing Benchmarks
+
+If you're familiar with the Java Microbenchmark Harness (JMH) toolkit, you'll find that the `kotlinx-benchmark`
+library shares a similar approach to crafting benchmarks. This compatibility allows you to seamlessly run your
+JMH benchmarks written in Kotlin on various platforms with minimal, if any, modifications.
+
+Like JMH, kotlinx-benchmark is annotation-based, meaning you configure benchmark execution behavior using annotations.
+The library then extracts metadata provided through annotations to generate code that benchmarks the specified code
+in the desired manner.
+
+To get started, let's examine a simple example of a multiplatform benchmark:
+
+```kotlin
+import kotlinx.benchmark.*
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 20, time = 1, timeUnit = TimeUnit.SECONDS)
```

Comment:
I think it's worth using some values different from what is passed to `@Masurement` to show that these two annotations don't have to be in sync.

Reply:
Agree. Done.